From e6e526b9911a1236ec27750c454079ebe1371d69 Mon Sep 17 00:00:00 2001
From: Steven Lee <steven_lee@aspeedtech.com>
Date: Mon, 6 Feb 2023 15:40:11 +0800
Subject: [PATCH] Support cpp 20

The latest openbmc sdbusplus library uses c++20 standard
mctpd should also use c++20 as it has dependency with sdbusplus.

Signed-off-by: Steven Lee <steven_lee@aspeedtech.com>
---
 CMakeLists.txt       | 2 +-
 src/SMBusBinding.cpp | 5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7ad941c..494f834 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@ project(mctpd CXX)
 option(BUILD_STANDALONE "Use outside of YOCTO depedencies system" OFF)
 
 set(BUILD_SHARED_LIBRARIES OFF)
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 add_definitions(-DMCTP_ASTPCIE_RESPONSE_WA)
diff --git a/src/SMBusBinding.cpp b/src/SMBusBinding.cpp
index dd61f15..e04916f 100644
--- a/src/SMBusBinding.cpp
+++ b/src/SMBusBinding.cpp
@@ -785,7 +785,8 @@ std::string SMBusBinding::SMBusInit()
 void SMBusBinding::readResponse()
 {
     smbusReceiverFd.async_wait(
-        boost::asio::posix::descriptor_base::wait_error, [this](auto& ec) {
+        boost::asio::posix::descriptor_base::wait_error,
+	[this](const boost::system::error_code& ec) {
             if (ec)
             {
                 phosphor::logging::log<phosphor::logging::level::ERR>(
@@ -1380,4 +1381,4 @@ void SMBusBinding::updateRoutingTableEntry(
         sizeof(smbusData->slave_addr);
 
     routingTable.updateEntry(entry.routeEntry.routing_info.starting_eid, entry);
-}
\ No newline at end of file
+}
-- 
2.25.1

